From bee53a0159d2c892d651e99c57fbf2e35a5173a6 Mon Sep 17 00:00:00 2001
From: Tianmi Chen <tianmi.chen@intel.com>
Date: Wed, 22 May 2013 13:56:41 +0800
Subject: libstagefright: implement error concealment for I/O error

BZ: 107755

Implement error concealment for I/O error to avoid system crash.

Category:aosp improvement
Domain: Video.Media-container
Origin: internal
Upstream-Candidate: yes

Change-Id: Ie938c0ad0dc658127a688865efae1daf2e5952d9
Orig-Change-Id: I1e02d2f9308fefcb91a313f014e49d28ce885e58
Signed-off-by: Tianmi Chen <tianmi.chen@intel.com>
Signed-off-by: ywan171 <yi.a.wang@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../libstagefright/matroska/MatroskaExtractor.cpp  |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/matroska/MatroskaExtractor.cpp b/media/libstagefright/matroska/MatroskaExtractor.cpp
index c450db3..9acb1cc 100755
--- a/media/libstagefright/matroska/MatroskaExtractor.cpp
+++ b/media/libstagefright/matroska/MatroskaExtractor.cpp
@@ -278,6 +278,16 @@ void BlockIterator::advance_l() {
 
             res = mCluster->Parse(pos, len);
             ALOGV("Parse (2) returned %ld", res);
+
+            if (res < 0) {
+                // I/O error
+
+                ALOGE("Cluster::Parse (2) returned result %ld", res);
+
+                mCluster = NULL;
+                break;
+            }
+
             CHECK_GE(res, 0);
 
             mBlockEntryIndex = 0;
-- 
1.7.4.1

